unittests/BuildAnalysis/CriticalPathSolverTests.swift

```diff
@@ -0,0 +1,145 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+import XCTest
+// The Swift package has llbuildSwift as module
+#if SWIFT_PACKAGE
+import llbuild
+import llbuildSwift
+#else
+import llbuild
+#endif
+import llbuildAnalysis
+
+class CriticalPathSolverTests: XCTestCase {
+    struct Input {
+        let keys: [BuildKey]
+        let lookup: [BuildKey: RuleResult]
+        let keyByName: [String: BuildKey]
+        
+        func path(with names: [String]) throws -> CriticalPath? {
+            let expectedKeys = names.compactMap { keyByName[$0] }
+            let rules = expectedKeys.compactMap { lookup[$0] }
+            guard expectedKeys.count == names.count, rules.count == names.count else { return nil }
+            return try CriticalPath.construct(Array(zip(expectedKeys, rules)))
+        }
+    }
+    
+    typealias InputData = [(key: String, deps: [String], start: Double, end: Double)]
+    static let exampleData: InputData = [
```

Comment:
Can this be inlined into the tests that use it?

Reply:
Ha, didn't realize there's actually just one test using that :)